unit_test/fb_serializer_test.cc

```diff
@@ -137,3 +136,35 @@ BOOST_AUTO_TEST_CASE(fb_serializer_generic_event_content_encoding) {
   const auto& batch_metadata = *(event_batch->metadata());
   BOOST_CHECK_EQUAL(batch_metadata.content_encoding()->c_str(), value::CONTENT_ENCODING_DEDUP);
 }
+
+BOOST_AUTO_TEST_CASE(fb_serializer_generic_event_metadata) {
+  data_buffer db;
+  fb_collection_serializer<generic_event> collection_serializer(db, value::CONTENT_ENCODING_DEDUP, "app_id");
+  const char* event_id("event_id");
+  const timestamp ts;
+  cb_serializer serializer;
+  ranking_response rr(event_id);
+  rr.set_model_id("model_id");
+  rr.push_back(1, 0.2);
+  rr.push_back(0, 0.8);
+
+  auto buffer = serializer.event("my_context", action_flags::DEFERRED, v2::LearningModeType_Apprentice, rr);
+
+  generic_event ge(event_id, ts, v2::PayloadType_CB, std::move(buffer), event_content_type::IDENTITY);
+  collection_serializer.add(ge);
+  BOOST_CHECK_EQUAL(reinforcement_learning::error_code::success, collection_serializer.finalize(nullptr));
+
+  flatbuffers::Verifier v(db.body_begin(), db.body_filled_size());
+  const v2::EventBatch *event_batch = v2::GetEventBatch(db.body_begin());
+  BOOST_CHECK(event_batch->Verify(v));
+
+  const auto& events = *(event_batch->events());
+
+  for (size_t i = 0; i < events.size(); ++i) {
+    const auto& serialized_event = *events[(flatbuffers::uoffset_t)i];
+    const auto& payload = serialized_event.payload();
+    const v2::Event *event = flatbuffers::GetRoot<v2::Event>(payload);
+    //const auto& metadata = *(event->meta());
+    //BOOST_CHECK_EQUAL(metadata.app_id()->c_str(), "app_id");
```

Comment:
@ataymano @olgavrou @kumpera - This pr adds app_id to the event metadata for v2. 

I am running into an issue when adding test for checking metadata. 

Line BOOST_CHECK_EQUAL(metadata.app_id()->c_str(), "app_id"); in the test throws below exception. Any ideas why that could happen. 
**Message: memory access violation occurred at address 0x914539aa, while attempting to  read inaccessible data**

Reply:
Actually I get the memory access violation for any metadata field not just app_id.

Reply:
looking